### DISCUSSION TITLE
Node regression dtype error in .backward()

### DISCUSSION THREAD

**Discussion by bhanu1106** [pyg-team/pytorch_geometric#5142] [Q&A] [answered]
I wanted to do a simple node regresssion problem , so i used coral dataset from pytorch geometric and generated random values for target variable . On this toy dataset, I had tried to do a node regression using torch.nn.MSELoss() loss function . Now I am facing dtype error in .backward() as follows : `RuntimeError: Found dtype Double but expected Float` . I even tried to change the data type from float64 to float32 . But facing the same error . 
![Screenshot from 2022-08-05 11-16-16](https://user-images.githubusercontent.com/82653939/183010163-f53103e0-295d-4cf3-a88e-7ef3d8adcfa1.png)
![Screenshot from 2022-08-05 11-17-44](https://user-images.githubusercontent.com/82653939/183010185-727ba30b-a76c-4a37-bcc4-f863ebdd51cd.png)

**Comment by rusty1s** [comment, marked as answer]
You should check that `out.dtype` and `data.y.dtype` match. Can you confirm?

**Reply by bhanu1106** [reply]
Thank you very much for the reply . changing data.y.dtype from float64 to float32 solved the problem
